app/uk/gov/hmrc/tradergoodsprofiles/controllers/CreateRecordController.scala

```diff
@@ -44,22 +42,16 @@ class CreateRecordController @Inject() (
 
   def createRecord(eori: String): Action[JsValue] =
     (authAction(eori) andThen validateHeaderAction).async(parse.json) { implicit request =>
-      (for {
-        createRequest <- validateCreateRecordRequest(request.body)
-        response      <- routerService.createRecord(eori, createRequest)
-      } yield Created(Json.toJson(response))).merge
+      validateRequestBody[APICreateRecordRequest](request.body, uuidService) match {
```

Comment:
you may want to split this into function to be more clear. E.g. 

`def requestAccreditation(eori: String, recordId: String): Action[JsValue] =
    (authAction(eori) andThen validateHeaderAction).async(parse.json) { implicit request =>
      validateRequestBody[RequestAccreditationRequest](request.body, uuidService) match {
        case Left(errorResponse)        => Future.successful(BadRequest(Json.toJson(errorResponse)))
        case Right(updateRecordRequest) => sendAccreditation(eori, recordId, updateRecordRequest)
      }
    }

  private def sendAccreditation(
    eori: String,
    recordId: String,
    accreditationRequest: RequestAccreditationRequest
  )(implicit hc: HeaderCarrier): Future[Result] =
    routerService.requestAccreditation(eori, recordId, accreditationRequest).map {
      case Left(serviceError) =>
        Status(serviceError.status)(toJson(serviceError.errorResponse))
      case Right(_)           =>
        Created
    }`